Problem Statement - 4/SerializationTest.java

```diff
@@ -0,0 +1,197 @@
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Scanner;
+
+// A address class --
+
+class Address implements Serializable {
+
+    // explicitly declare and initialize serialVersionUID
+    private static final long serialVersionUID = 652;
+    private String city;
+    private String state;
+    private String country;
+    private int pinCode;
+
+    // Getters and Setters
+    public String getCity() {
+        return city;
+    }
+
+    public void setCity(String city) {
+        this.city = city;
+    }
+
+    public String getState() {
+        return state;
+    }
+
+    public void setState(String state) {
+        this.state = state;
+    }
+
+    public String getCountry() {
+        return country;
+    }
+
+    public void setCountry(String country) {
+        this.country = country;
+    }
+
+    public int getPinCode() {
+        return pinCode;
+    }
+
+    public void setPinCode(int pinCode) {
+        this.pinCode = pinCode;
+    }
+
+    // Constructor
+    Address(String city, String state, String country, int pinCode) {
+        this.city = city;
+        this.country = country;
+        this.state = state;
+        this.pinCode = pinCode;
+    }
+}
+
+// A Student class --
+
+class Student implements Serializable {
+
+    // Solution for Invalid Class Exception
+
+    /*
+     * To explicitly declare and initialize serialVersionUID .
+     * If a class does not explicitly define a private static final long
+     * serialVersionUID
+     * in the code it will be autogenerated, and there is no guarantee that
+     * different machines
+     * will generate the same id; it looks like that is exactly what happened. Also
+     * if the
+     * classes are different in any way (using different versions of the class) the
+     * autogenerated serialVersionUIDs will also be different.
+     */
+
+    // private static final long serialVersionUID = 652;
+    String firstName;
+
+    // Initial
+    // Object dateOfBirth;
+
+    // After Change
+    Object dateOfBirth;
+    Address address;
+
+    // iniatilizing simple date object for formatting
+    SimpleDateFormat simpleDate = new SimpleDateFormat("dd/mm/yyyy"); // for date formatting
+
+    // Constructor
```

Comment:
obvious commenting you should have to avoid , comment at high level way like what is intention for doing thing in particular code block(code block which is not easily understandable clearly ,there you have to comment) .